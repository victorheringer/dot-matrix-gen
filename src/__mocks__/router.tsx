import { MemoryRouter } from "react-router-dom";

export function RouterWrapper({ children }: any) {
  return <MemoryRouter>{children}</MemoryRouter>;
}
