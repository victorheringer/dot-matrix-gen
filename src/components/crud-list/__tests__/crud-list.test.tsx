import { render, fireEvent } from "@testing-library/react";
import { RouterWrapper } from "../../../__mocks__/router";
import {
  CrudList,
  CrusListItemTitle,
  CrudListItemLink,
  CrudListItemColumn,
  CrudListItemGrid,
} from "../index";

describe("CrudList", () => {
  it("should render correctly with content", () => {
    const tree = render(
      <CrudList
        title="test-title"
        createBtnText="test-btn-text"
        onCreateClick={jest.fn()}
        total={10}
      >
        <p>test</p>
      </CrudList>
    ).asFragment();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly without content", () => {
    const tree = render(
      <CrudList
        title="test-title"
        createBtnText="test-btn-text"
        onCreateClick={jest.fn()}
        total={0}
      >
        <p>test</p>
      </CrudList>
    ).asFragment();
    expect(tree).toMatchSnapshot();
  });

  it("should correctly call createBtnText on press", () => {
    const mockOnPress = jest.fn();

    const { getByText } = render(
      <CrudList
        title="test-title"
        createBtnText="test-btn-text"
        onCreateClick={mockOnPress}
        total={0}
      >
        <p>test</p>
      </CrudList>
    );

    fireEvent.click(getByText("New"));
    expect(mockOnPress).toBeCalledTimes(1);
  });
});

describe("CrusListItemTitle", () => {
  it("should render correctly", () => {
    const tree = render(
      <CrusListItemTitle>test</CrusListItemTitle>
    ).asFragment();
    expect(tree).toMatchSnapshot();
  });
});

describe("CrudListItemLink", () => {
  it("should render correctly", () => {
    const tree = render(
      <RouterWrapper>
        <CrudListItemLink to="test/url">test</CrudListItemLink>
      </RouterWrapper>
    ).asFragment();
    expect(tree).toMatchSnapshot();
  });
});

describe("CrudListItemColumn", () => {
  it("should render correctly", () => {
    const tree = render(
      <CrudListItemColumn>test</CrudListItemColumn>
    ).asFragment();
    expect(tree).toMatchSnapshot();
  });
});

describe("CrudListItemGrid", () => {
  it("should render correctly", () => {
    const tree = render(<CrudListItemGrid>test</CrudListItemGrid>).asFragment();
    expect(tree).toMatchSnapshot();
  });
});
