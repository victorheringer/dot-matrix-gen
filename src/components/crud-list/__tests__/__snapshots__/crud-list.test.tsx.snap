// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CrudList should render correctly with content 1`] = `
<DocumentFragment>
  <div
    class="sc-ksZaOG jPhuDQ"
  >
    <h1
      class="sc-evZas jgMWAF"
    >
      test-title
    </h1>
    <div
      class="sc-hAZoDl duqJcE"
    >
      <div
        class="sc-jSMfEi gwRlBM"
      >
        <div
          class="sc-fnykZs knWTCK"
        >
          <div
            class="sc-fEOsli hVvMIe"
          >
            <p
              class="sc-idiyUo kXqhxf"
            >
              test-btn-text
            </p>
          </div>
          <button
            class="sc-hKMtZM hlCIed"
          >
            New
          </button>
        </div>
      </div>
      <p>
        test
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CrudList should render correctly without content 1`] = `
<DocumentFragment>
  <div
    class="sc-ksZaOG jPhuDQ"
  >
    <h1
      class="sc-evZas jgMWAF"
    >
      test-title
    </h1>
    <div
      class="sc-hAZoDl duqJcE"
    >
      <div
        class="sc-jSMfEi gwRlBM"
      >
        <div
          class="sc-fnykZs knWTCK"
        >
          <div
            class="sc-fEOsli hVvMIe"
          >
            <p
              class="sc-idiyUo kXqhxf"
            >
              test-btn-text
            </p>
          </div>
          <button
            class="sc-hKMtZM hlCIed"
          >
            New
          </button>
        </div>
      </div>
      <p>
        test
      </p>
      <p
        class="sc-bjUoiL caxqdG"
      >
        You don't have any content yet, just click above and create a new one!
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CrudListItemColumn should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-dmRaPn iKIcXE"
  >
    test
  </div>
</DocumentFragment>
`;

exports[`CrudListItemGrid should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-kgflAQ fCPQaH"
  >
    test
  </div>
</DocumentFragment>
`;

exports[`CrudListItemLink should render correctly 1`] = `
<DocumentFragment>
  <a
    class="sc-hHLeRK hAxDos"
    href="/test/url"
  >
    test
  </a>
</DocumentFragment>
`;

exports[`CrusListItemTitle should render correctly 1`] = `
<DocumentFragment>
  <p
    class="sc-dIouRR TTRYl"
  >
    test
  </p>
</DocumentFragment>
`;
